### PR TITLE
106 : Validate end date greater than start date in new invoice screen

### PR DESCRIPTION
**Pull request by ducmanh2111**
http://redmine.dtpebike.vn/issues/106

### CODE REVIEW

File: app/forms/contracts_form.rb [hunk 1]
```diff
@@ -57,4 +61,11 @@ def destroy
   def contract
     @contract ||= record || Contract.new
   end
+
+  def ended_date_must_greater_than_started_date
+    return true unless Date.parse(started_date) > Date.parse(ended_date)
```
Comment (ngocnguyenbk) [inline app/forms/contracts_form.rb:66]: Viết cho thuận đi anh.

File: spec/forms/contracts_form_spec.rb [hunk 2]
```diff
@@ -16,6 +16,22 @@
     it { should validate_presence_of(:room_price).with_message(:blank) }
     it { should validate_presence_of(:started_date).with_message(:blank) }
     it { should validate_presence_of(:ended_date).with_message(:blank) }
+    context "when ended_date before started date" do
```
Comment (ngocnguyenbk) [inline spec/forms/contracts_form_spec.rb:19]: Viết trong hàm save

Comment (ducmanh2111) [reply spec/forms/contracts_form_spec.rb]: OK

**Review by ngocnguyenbk — APPROVED** [review]

**Comment by ngocnguyenbk** [comment]
LGTM

**Pull request merged**